floto/static/overview.js

```diff
@@ -0,0 +1,28 @@
+const { createApp, ref, reactive, watch, onMounted, watchEffect } = Vue
+
+createApp({
+    delimiters: ["[[", "]]"],
+    setup() {
+        const devices = reactive({"value": []})
+        let devices_loading = ref("primary")
+        fetch_with_retry("/api/devices", callback=function(json){
+        devices.value = json
+        devices_loading.value = false
+        })
+        onMounted(() => {
+            var map = L.map('map').setView([38.178318, -101.0093573], 4);
+            L.tileLayer('https://tile.openstreetmap.org/{z}/{x}/{y}.png', {
+                maxZoom: 19,
+                attribution: '© OpenStreetMap contributors'
+            }).addTo(map);
+            // watchEffect to reactively add markers once the data is available.
+            watchEffect(() => {
+                devices.value.forEach(device => {
+                    if (device.latitude && device.longitude && device.latitude !== 0.0 && device.longitude !== 0.0) {
+                    L.marker([device.latitude, device.longitude]).addTo(map)
+                        .bindPopup(`<b>${device.device_name}</b>`);
```

Comment:
Can you add the status online/offline as well? I see there is not a simple way to change the marker color, we'd have to create different color icons images.

Reply:
Actually it looks like there is a DivIcon class we could use, or here I found some people who made custom images https://stackoverflow.com/questions/23567203/leaflet-changing-marker-color

Reply:
We could use DivIcon and nest inside it a [Quasar Icon](https://quasar.dev/vue-components/icon) which has support for [material design icons](https://fonts.google.com/icons?icon.set=Material+Icons&selected=Material+Icons+Outlined:place:&icon.query=point) and colors.